Modules/Sources/DesignSystem/Components/Content Preview/ContentPreview.swift

```diff
@@ -53,32 +55,56 @@ public struct ContentPreview: View {
         }
     }
 
-    private func image(url: URL) -> some View {
-        AsyncImage(url: url) { phase in
+    private func image(configuration: ImageConfiguration) -> some View {
+        AsyncImage(url: configuration.url) { phase in
             Group {
                 switch phase {
                 case .success(let image):
                     image.resizable()
                 default:
-                    Color.DS.Background.tertiary
+                    if let placeholder = configuration.placeholder {
+                        placeholder
+                            .resizable()
+                    } else {
+                        Color.DS.Background.tertiary
+                    }
                 }
             }
             .frame(width: imageSize, height: imageSize)
             .clipShape(Circle())
         }
     }
+
+    // MARK: - Types
+
+    public struct ImageConfiguration {
+        let url: URL?
+        let placeholder: Image?
+
+        public init(url: URL?, placeholder: Image? = nil) {
+            self.url = url
+            self.placeholder = placeholder
+        }
+
+        public init(url: String?, placeholder: Image? = nil) {
+            self.init(url: URL(string: url ?? ""), placeholder: placeholder)
```

Comment:
Thanks for the feedback @justtwago. That is precisely what's happening because URL(string:) returns nil if the url is invalid. So URL(string: "") returns nil. See [Apple documentation](https://developer.apple.com/documentation/foundation/url/3126806-init).

Reply:
Oh, good to know! That's really convenient to have a nullable init function! 🤯